test/post.test.js

```diff
@@ -282,4 +288,23 @@ describe('Post data class tests', () => {
       await expect(Post.createFromArticle(article, feed)).rejects.toThrow();
     });
   });
+
+  describe('Post.createFromArticle() with youtube feeds tests', () => {
+    let articles;
+    beforeEach(async () => {
+      nockRealWorldYouTubeFeedResponse();
+      articles = await parse.parseURL(getRealWorldYouTubeFeedUri());
+      expect(Array.isArray(articles.items)).toBe(true);
+      expect(articles.items.length).toBe(15);
+    });
+
+    test('Post.createFromArticle() should create Post with YouTube video article', async () => {
+      const article = articles.items[0];
+      const id = await Post.createFromArticle(article, feed);
+      const videoPost = await Post.byId(id);
+
+      expect(videoPost.title).toBe('DPS909 OSD600 Week 03 - Fixing a Bug in the Azure JS SDK');
+      expect(videoPost.url).toBe('https://www.youtube.com/watch?v=mNuHA7vH6Wc');
```

Comment:
Worth checking `type` here too?

Reply:
Since this is in the "backend" level, and not the API level, there is no usage for `type`. I didn't put the `determinePostType` function in the `backend` because it was only used in the `posts` microservice. If we want, I can also place a copy of `determinePostType` here too, although it will only be used for the test.